src/helpers/randomizeCard/index.js

```diff
@@ -0,0 +1,736 @@
+import { FACTIONS, RACES, TYPES, RARITIES } from '~/constants/game'
+import arrayRandom from '~/helpers/arrayRandom'
+
+//##############
+//# To-Do List #
+//##############
+
+// Add more effects
+// Add effects that target structures
+// Make Satyr self-base destruction less prevalent
+// Add Dragons, Pirates, Felines, and plain Elders/Ancients
+// Add better filters - maybe even filter for effect classes?
+// Add effect classes for different types of effects
+// Add randomly generated names - maybe combo off of effect classes
+// Fix inconsistencies:
+// -- Affecting bordering/surrounding units with no bordering/surrounding units
+// -- Spawning units behind when played bordering the base
+// -- Tweak Convert targeting/conditions a bit - add a new category for Convert targets, and increase the number of conditions
+// -- Fix unspend mana calculations
+
+//################################
+//# Settings to mess around with #
+//################################
+
+// Turns on verbose logging, displaying how each card is calculated.
+var verbose = false 
+
+// Removes the caps on minimum mana, maximum mana, and effect cost. Can make some funy units.
+var uncapped = false 
+
+// These parameters can be used to limit the types of cards generated.
+// maxEffCost can be used to tweak the maximum power of the effect.v
+// Higher values tend towards smaller units with more potent effects.
+// Be sensible with the faction and tribe requirments. Don't put in anything impossible.
+var minMana = 1
+var maxMana = 9
+var maxEffCost = 4.5
+var faction = ""
+var tribe = ""
+var cardsToGenerate = 1
+
+//#####################
+//# Class definitions #
+//#####################
+
+class Tribe {
+	constructor(name,faction,movRange) {
+		this.name = name
+		this.faction = faction
+		this.movRange = movRange
+		this.effects = []
+	}
+	addEffect(effect) {
+		this.effects.push(effect)
+	}
+}
+
+class Effect {
+	constructor(desc,valid,cost) {
+		this.desc = desc
+		this.valid = valid
+		this.cost = cost
+	}
+}
+
+class Slot {
+	constructor(type,part1,part2,cost) {
+		this.type = type
+		this.part1 = part1
+		this.part2 = part2
+		this.cost = cost
+	}
+}
+ 
+function eff(tribeName, effect) {
+	tribes.forEach((tribe) => {
+		if(tribe.name == tribeName) {
+			tribe.addEffect(effect)
+		}
+	})
+}
```

Comment:
Can be simplified with `Array.prototype.filter`. Also maybe we give it a more explicit name? 😅 
```suggestion
function eff(tribeName, effect) {
  tribes
    .filter(tribe => tribe.name === tribeName)
    .forEach(tribe => tribe.addEffect(effect))
```

Reply:
Since I was typing it out a lot, I thought it would be slightly easier to give it a short name.